kotest-framework/kotest-framework-engine/src/jvmTest/kotlin/com/sksamuel/kotest/engine/test/timeout/ContainerTimeoutTest.kt

```diff
@@ -1,38 +1,42 @@
 package com.sksamuel.kotest.engine.test.timeout
 
 import io.kotest.assertions.asClue
-import io.kotest.common.testTimeSource
+import io.kotest.core.config.ProjectConfiguration
 import io.kotest.core.spec.style.FunSpec
+import io.kotest.datatest.withData
 import io.kotest.engine.TestEngineLauncher
 import io.kotest.engine.listener.CollectingTestEngineListener
 import io.kotest.matchers.collections.shouldContainExactly
 import io.kotest.matchers.nulls.shouldNotBeNull
 import io.kotest.matchers.shouldBe
-import kotlinx.coroutines.delay
+import io.kotest.matchers.string.shouldStartWith
 import kotlin.time.Duration.Companion.milliseconds
-import kotlin.time.measureTime
 
 class ContainerTimeoutTest : FunSpec() {
    init {
-      coroutineTestScope = true
+      context("container test should timeout if nested exceeds parent timeout") {
+         withData(
+            nameFn = { (coroutineTestScope, message) -> "coroutineTestScope = $coroutineTestScope -> $message" },
+            false to "Test 'a' did not complete within 100ms",
+            true to "After waiting for 100ms, the test coroutine is not completing, there were active child jobs",
+         ) { (enableCoroutineTestScope, message) ->
+            val collector = CollectingTestEngineListener()
+            val config = ProjectConfiguration().apply {
+               coroutineTestScope = enableCoroutineTestScope
+            }
 
-      test("container test should timeout if nested exceeds parent timeout") {
-         val collector = CollectingTestEngineListener()
-
-         val duration = testTimeSource().measureTime {
             TestEngineLauncher(collector)
+               .withConfiguration(config)
                .withClasses(NestedTimeout::class)
-               .async()
-         }
-
-         duration shouldBe 100.milliseconds
```

Comment:
After removing `coroutineTestScope` from the Spec itself, duration is not deterministic anymore, that's why I decided to remove this assertion.